flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/nodes/dataset/BatchTableSourceScan.scala

```diff
@@ -32,18 +31,20 @@ class BatchTableSourceScan(
     cluster: RelOptCluster,
     traitSet: RelTraitSet,
     table: RelOptTable,
-    rowType: RelDataType)
-  extends BatchScan(cluster, traitSet, table, rowType) {
+    val tableSource: BatchTableSource[_])
```

Comment:
Do we have to declare this field as `val` ? It seems that it is only used in this class.